pkg/controller/propagator/propagation.go

```diff
@@ -676,22 +685,37 @@ func (r *ReconcilePolicy) processTemplates(replicatedPlc *policiesv1.Policy, dec
 			reqLogger.Error(tplErr, "Failed to resolve templates")
 
 			r.recorder.Event(rootPlc, "Warning", "PolicyPropagation",
-				fmt.Sprintf("Failed to resolve templates for policy %s/%s for cluster %s/%s .", rootPlc.GetName(), rootPlc.GetNamespace(), decision.ClusterNamespace, decision.ClusterName))
+				fmt.Sprintf("Failed to resolve templates for cluster %s/%s: %s", decision.ClusterNamespace, decision.ClusterName, tplErr.Error()))
+
+			//Set an annotation on the policyTemplate(e.g. ConfigurationPolicy)  to the template processing error msg
+			//managed clusters will use this when creating a violation
+			policyTObjectUnstructured := &unstructured.Unstructured{}
+			jsonErr := json.Unmarshal(policyT.ObjectDefinition.Raw, policyTObjectUnstructured)
```

Comment:
Could you please use the `json` package from Kubernetes? It performs special handling of certain types such as JSON numbers. It has the same interface as `encoding/json`.
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/json

Reply:
fixed